cip/1.accepted/CIP2019-02-06-SET.adoc

```diff
@@ -0,0 +1,227 @@
+For this reason, there are multiple semantically identical ways to remove a property from a node or relationship:
+
+* Assign the property to null
+** `SET n.prop = null`
+** `SET n.prop = expression`  where expression evaluates to null, for example a variable or a parameter
+* Remove the property: `REMOVE n.prop`
+* Assign the node or relationship to a Map value that does not contain the previously existing property: `SET n = {a: var, b:var}`
+
+====== Adding or Removing Labels
+
+Labels can only be added to or removed from nodes.
+
+* Adding one label to a node: `SET n:Foo`
+* Adding multiple labels to a node: `SET n:Foo:Bar`
+* Removing one label from a node: `REMOVE n:Foo`
+* Removing multiple labels from a node: `REMOVE n:Foo:Bar`
+
+== Interaction with existing features
+
+=== Deleted Elements
+
+The `PropertyExpression` or `Variable` on the left-hand-side of a `SetItem` or `RemoveItem` might refer to a node or relationship that has been deleted previously in the same query.
+In this case, the `SET` or `REMOVE`  is a no-operation.
+
+=== Order Dependencies
+
+There are cases where the order of the incoming records can influence the final results.
+This could lead to unexpected behaviour.
+Consider the following example:
+[source, cypher]
+----
+MATCH (p1:Product {pid:85}), (p2:Product {pid:125})
+SET p1.name = p2.name
+----
+
+If there is more than one product with `pid=125`, then the `SET` clause will be applied twice to each `p1`, with a different `p2`.
+If all the `p2.name` values for `p2.pid=125` are the same, this has no surprising result.
+However, if they are not the same, the last one set will be the one that persists.
+This means the final result is a function of the order in which the `p2` nodes are found.
+This order is likely not something the user is aware of or in control of, and therefore could lead to surprising behaviour.
+
+==== Planner Warnings
+
+It is recommended that query planners detect cases where the query could result in order dependencies and issue warnings that describe this case to allow developers and users to take this into account when writing queries.
+The conditions to look for would be:
+
+* We have a `SET` on an element property where it is not known if the same element will occur on multiple rows or not.
+** If the elements come from a unique index, and their cardinality is never increased (eg. expand in pattern expression or cartesian product with another match), then the element is not repeated.
+** If the user uses a `WITH DISTINCT`, or an aggregation that groups on the element before the set to ensure uniqueness, the element is not repeated.
+* And the right-hand side expression is not guaranteed to generate the same value for each case of the same left-hand side.
+** Literal expressions are safe.
+** Expressions that are only functions of the left-hand side element are safe.
+
+==== Runtime Errors
```

Comment:
It's a bit weird that this CIP allows three different levels of enforcement: no warnings, warnings, or errors, and that these are all listed as compliant implementations.